File: leetcode/0001-1000/0392-is-subsequence/IsSubsequence.java
public class IsSubsequence {

  public static void main(String[] args) {
    System.out.println(new Solution().isSubsequence("abc", "ahbgdc"));
  }
}

class Solution {
  public boolean isSubsequence(String s, String t) {
    int i = 0;
    int j = 0;
    int n = s.length();
    int m = t.length();

    while (i < n && j < m) {
      if (s.charAt(i) == t.charAt(j)) {
        i++;
      }

      j++;
    }

    return i == n;
  }
}

File: leetcode/0001-1000/0301-0400/0391-0340/0392-is-subsequence/IsSubsequence.ts
function isSubsequence(s: string, t: string): boolean {
  let i = 0;
  let j = 0;

  while (i < s.length && j < t.length) {
    if (t[j] === s[i]) {
      i++;
    }

    j++;
  }

  return i === s.length;
};

console.log(isSubsequence("abc", "ahbgdc"));
console.log(isSubsequence("aa", "ab"));
console.log(isSubsequence("axc", "ahbgdc"));

